packages/workspace/src/utilities/generate-globs.ts

```diff
@@ -52,6 +52,10 @@ export function createGlobPatternsForDependencies(
       projectGraph
     );
 
+    // change cwd to support scripts running from project folder, mainly nx:run-commands with cwd option
+    const initialCwd = process.cwd();
+    if (initialCwd !== workspaceRoot) process.chdir(workspaceRoot);
```

Comment:
Rather than changing `process.cwd`, we should be able to resolve the paths relative to workspaceRoot. The changeset would probably be a bit larger, but I think it would be preferred.

Reply:
I would agree. I initially went on that track but stumbled across issues that seemed to go deeper than the scope of `globPatternforDependencies`, possibly in the `ignore` library. But i'll revisit!

Reply:
IIRC `ignore` requires that the paths are relative. Might need to resolve, and then make relative rather than just going w/ the fully resolved paths.

Reply:
Changed implementation as suggested.
I was able to solve my original issue with the `ignore` lib which stemmed from the 5.0.0 update to the library.
See: https://github.com/kaelzhang/node-ignore/issues/20 for context.
I was trying to pass relative paths as `../foo/bar` which it does not accept and the thrown error that was a bit misguiding:
`path should be a path.relative()d string, but got "../foo/bar"` and in my world `../foo/bar` is a "relative()d" string :)